QuickShiftUI/Components/Pages/Shift.razor

```diff
@@ -0,0 +1,67 @@
+﻿@page "/MyShifts/{Id:long}"
+@using DTOs.Shift
+@using QuickShiftUI.Services
+@inject IShiftService ShiftService
+@inject NavigationManager Navigation
+@inject AuthenticationStateProvider AuthProvider
+
+
+<h3>Shift details</h3>
+
+<AuthorizeView>
+
+    <Authorized>
+        @if (displayedShift != null)
+        {
+            <div class="card mt-4">
+                <div class="card-header">
+                    <h4>@displayedShift.TypeOfShift</h4>
+                </div>
+
+                <div class="card-body">
+                    <p>
+                        <b>Start time: </b><span>@displayedShift.StartDateTime</span>
+                        <br/>
+                        <b>End time: </b><span>@displayedShift.EndDateTime</span>
+                    </p>
+
+                    <p><b>Location: </b> <span>@displayedShift.Location</span></p>
+
+
+                    <p><b>Status: </b> <span>@displayedShift.ShiftStatus</span> </p>
+
+                    <p>
+                        <b>Shift description:</b>
+                        <br>
+                        <span>@displayedShift.Description</span>
+                    </p>
+                </div>
+
+            </div>
+        }
+    </Authorized>
+
+    <NotAuthorized> Log in to see the shift! </NotAuthorized>
+
+</AuthorizeView>
+
+<button class="btn btn-secondary mt-4" @onclick="BackToShifts">Return</button>
+
+@code {
+    [Parameter] public long Id { get; set; }
+    public ShiftDTO? displayedShift { get; set; }
+
+    [CascadingParameter] public Task<AuthenticationState> State { get; set; }
+    private int loggedInUserId;
+
+    protected override async Task OnParametersSetAsync()
+    {
+        displayedShift = await ShiftService.GetShiftByIdAsync(Id);
```

Comment:
This needs some kind of error-handling. What happens if somebody changes the Id of the shift in the navbar? It's just not gonna display anything in that case, but we should instead give some feedback to the user

Reply:
Now if somebody enters id of not existing shift it goes back to list of shifts, may be improved in the future but for now it does the job